be/src/exec/arrow_to_starrocks_converter.cpp

```diff
@@ -779,181 +779,192 @@ struct ArrowListConverter {
         using ArrowArrayType = typename arrow::TypeTraits<T>::ArrayType;
         using OffsetsType = typename T::offset_type;
         auto* concrete_array = down_cast<const ArrowArrayType*>(layer);
-        auto* arrow_offsets_data = concrete_array->raw_value_offsets() + layer_start_idx;
+        auto* arrow_offsets_data = concrete_array->raw_value_offsets() + array_start_idx;
         auto arrow_base_offset = arrow_offsets_data[0];
         arrow_offsets_data += 1;
-        auto start_idx = offsets_layer->size() - 1;
-        offsets_layer->resize(offsets_layer->size() + layer_num_elements);
-        auto* offsets_data = &offsets_layer->get_data().front() + start_idx;
+        auto start_idx = col_offsets->size() - 1;
+        col_offsets->resize(col_offsets->size() + num_elements);
+        auto* offsets_data = &col_offsets->get_data().front() + start_idx;
         auto base_offset = offsets_data[0];
         offsets_data += 1;
-        offsets_copy<OffsetsType>(arrow_offsets_data, arrow_base_offset, layer_num_elements, offsets_data, base_offset);
-    }
-
-    static Status apply(const arrow::Array* array, size_t array_start_idx, size_t num_elements, size_t depth_limit,
-                        const TypeDescriptor* type_desc, Column* column, size_t column_start_idx,
-                        [[maybe_unused]] uint8_t* null_data, ArrowConvertContext* ctx) {
-        std::vector<const arrow::Array*> list_layers;
-        std::vector<std::pair<size_t, size_t>> layer_ranges;
-        list_layers.reserve(depth_limit + 1);
-        layer_ranges.reserve(depth_limit + 1);
-        Status status;
-        status =
-                flatten_and_verify_list(array, array_start_idx, num_elements, depth_limit, &list_layers, &layer_ranges);
-        if (!status.ok()) {
-            return status;
-        }
+        offsets_copy<OffsetsType>(arrow_offsets_data, arrow_base_offset, num_elements, offsets_data, base_offset);
+    }
 
-        std::vector<UInt32ColumnPtr> offsets_layers;
-        std::vector<NullColumnPtr> nulls_layers;
-        Column* last_column = nullptr;
-        const TypeDescriptor* last_type_desc = nullptr;
-        status = flatten_and_verify_column(column, type_desc, depth_limit, &offsets_layers, &nulls_layers, &last_column,
-                                           &last_type_desc);
-        if (!status.ok()) {
-            return status;
+    template <typename T>
+    static arrow::Array* get_list_array_child(const arrow::Array* array, size_t array_start_idx, size_t num_elements,
+                                              size_t* child_array_start_idx = nullptr,
+                                              size_t* child_array_num_elements = nullptr) {
+        using ArrowArrayType = typename arrow::TypeTraits<T>::ArrayType;
+        using OffsetsType = typename T::offset_type;
+        if (child_array_start_idx && child_array_num_elements) {
+            auto child_array = down_cast<const ArrowArrayType*>(array);
```

Comment:
can nested list  be LargeList or FixedSizeList?

Reply:
add FixedSizeList in new pr